compiler/options.nim

```diff
@@ -447,6 +447,13 @@ proc isDefined*(conf: ConfigRef; symbol: string): bool =
                             osDragonfly, osMacosx}
     else: discard
 
+proc getBackend*(conf: ConfigRef): string =
+  ## Get the backend derived from config files.
+  result = "c" # default
+  if isDefined(conf, "cpp"): return "cpp"
```

Comment:
@Araq I need your help here.. it is "almost" working.. but the `isDefined` approach to derive the backend specified in the config.nims is not working as I expected.

This PR works if I put this in my config.nims:

```nim
switch("define", "cpp")
```

It feels like a bit of hack. Is there a way to get the "cpp" value if I have this snippet in the config:

```nim
if getCommand() == "c":
  setCommand("cpp")
```

Reply:
Actually if I put that define switch in the config.nims, it's breaking something else which I don't understand:

```
Hint: used config file '/home/kmodi/sandbox/nim/std_vector/config.nims' [Conf]
Hint: used config file '/home/kmodi/usr_local/apps/6/nim/devel/config/nimdoc.cfg' [Conf]
/home/kmodi/usr_local/apps/6/nim/devel/lib/system/excpt.nim(570, 12) Error: only a 'ref object' can be raised
/home/kmodi/usr_local/apps/6/nim/devel/lib/system/excpt.nim(571, 52) Error: type mismatch: got <ref StdException>
but expected one of:
proc what(ex: StdException): cstring
  first type mismatch at position: 1
  required type for ex: StdException
  but expression 'e' is of type: ref StdException

expression: what(e)
/home/kmodi/usr_local/apps/6/nim/devel/lib/system/cgprocs.nim(18, 6) Error: implementation of 'system.nimGetProcAddr(lib: LibHandle, name: cstring) [declared in /home/kmodi/usr_local/apps/6/nim/devel/lib/system/cgprocs.nim(18, 6)]' expected
debug: cmd = nim c -r $1
debug: d.conf.getBackend() = cpp
debug: c = /home/kmodi/usr_local/apps/6/nim/devel/bin/nim cpp -r $1
debug: c2 = /home/kmodi/usr_local/apps/6/nim/devel/bin/nim cpp -r /tmp/kmodi/.cache/nim/std_vector_d/std_vector/std_vector_snippet_107.nim
Hint: /home/kmodi/usr_local/apps/6/nim/devel/bin/nim cpp -r /tmp/kmodi/.cache/nim/std_vector_d/std_vector/std_vector_snippet_107.nim [Exec]
Hint: used config file '/home/kmodi/usr_local/apps/6/nim/devel/config/nim.cfg' [Conf]
Hint: used config file '/home/kmodi/usr_local/apps/6/nim/devel/config/config.nims' [Conf]
Hint: used config file '/home/kmodi/.config/nim/config.nims' [Conf]
Hint: 32854 LOC; 0.030 sec; 37.715MiB peakmem; Debug build; /tmp/kmodi/.cache/nim/std_vector_d/std_vector/std_vector_snippet_107.nim proj; /tmp/kmodi/.cache/nim/std_vector_d/std_vector/std_vector_snippet_107 out [SuccessX]
Hint: /tmp/kmodi/.cache/nim/std_vector_d/std_vector/std_vector_snippet_107  [Exec]
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(2334, 20) template/generic instantiation of `format` from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(2314, 65) template/generic instantiation from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(1844, 11) Error: can raise an unlisted exception: ref TimeFormatParseError
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(2421, 8) template/generic instantiation of `parse` from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(2406, 22) template/generic instantiation from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(1844, 11) Error: can raise an unlisted exception: ref TimeFormatParseError
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(2439, 18) template/generic instantiation of `format` from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(2314, 65) template/generic instantiation from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/pure/times.nim(1844, 11) Error: can raise an unlisted exception: ref TimeFormatParseError
...
```